src/connection/server_support.c

```diff
@@ -1365,20 +1365,13 @@ css_init (THREAD_ENTRY * thread_p, char *server_name, int name_length, int port_
 
   css_Server_connection_socket = INVALID_SOCKET;
 
-  /*
-   * The packing of the server name and type is done by setting the first
-   * character to the numer related to the server type enum value, the rest of the
-   * buffer space is used to copy the server name.
-   */
-  buffer = (char) (get_server_type () + '0');
-  buffer.append (server_name, name_length);
-  conn = css_connect_to_master_server (port_id, buffer.c_str (), name_length + 1);
+  conn = css_connect_to_master_server (port_id, message_to_master, message_to_master_length);
```

Comment:
As far as I saw, the signature of the function `css_connect_to_master_server` is still
```
extern CSS_CONN_ENTRY *css_connect_to_master_server (int master_port_id, const char *server_name, int name_length);
```
and the function is used to perform extra checks using the said argument;
Maybe it works now by virtue of the fact that the message contains first the server name and then the appended server type value, but at least the name of the parameter should specify that it is the entire "message_to_master", not just the server_name.
Instead of the already packed message, pass around a structure containing
```
{
  server name,
  type
}
```
(and, possibly, extra info is needed as per `css_pack_server_name`) and only pack when dispatching to the wire.
Even before this patch, the argument passed to `css_init` is already the "packed_server_name", not the actual server name.

Reply:
You're right, server_name & name_length arguments are outdated. Actually, they were outdated even before the changes of this patch. The arguments were renamed as message_to_master & message_to_master_length.

Since css_connect_to_master_server is also used by the client and its implementation is very similar, I preferred to just change the argument name, instead of passing server name & type.